notification-portlet-webcomponents/notification-icon/src/components/NotificationIcon.vue

```diff
@@ -0,0 +1,110 @@
+<template>
+  <div class="notification-icon">
+    <dropdown id="_uid" no-caret :variant="variant">
+      <template slot="button-content">
+        <font-awesome-icon icon="bell"/>
+        {{ unreadCount }}
+      </template>
+      <dropdown-header>Notifications</dropdown-header>
+      <notification-item
+        v-for="notification in notifications"
+        :key="notification.id || notification.body"
+        :notification="notification"
+      ></notification-item>
+      <dropdown-item :href="seeAllNotificationsUrl">See All Notifications</dropdown-item>
+    </dropdown>
+  </div>
+</template>
+
+<script>
+import Vue from "vue";
+import AsyncComputed from "vue-async-computed";
+import Dropdown from "bootstrap-vue/es/components/dropdown/dropdown";
+import DropdownHeader from "bootstrap-vue/es/components/dropdown/dropdown-header";
+import DropdownItem from "bootstrap-vue/es/components/dropdown/dropdown-item";
+import { library } from "@fortawesome/fontawesome-svg-core";
+import { faBell } from "@fortawesome/free-solid-svg-icons/faBell";
+import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";
+import oidc from "@uportal/open-id-connect";
+import ky from "ky";
+import NotificationItem from "./NotificationItem";
+
+Vue.use(AsyncComputed);
+
+library.add(faBell);
+
+export default {
+  name: "NotificationIcon",
+  props: {
+    userInfoApiUrl: {
+      type: String,
+      default: "/uPortal/api/v5-1/userinfo"
+    },
+    notificationApiUrl: {
+      type: String,
+      default: "/NotificationPortlet/api/v2/notifications"
+    },
+    seeAllNotificationsUrl: {
+      type: String,
+      default: "/uPortal/p/notification"
+    },
+    debug: {
+      type: Boolean,
+      default: false
+    }
+  },
+  asyncComputed: {
+    notifications: {
+      async get() {
+        const { notificationApiUrl, debug } = this;
+
+        try {
+          const headers = debug
+            ? {}
+            : {
+                Authorization: "Bearer " + (await oidc()).encoded,
+                "content-type": "application/jwt"
+              };
+          return await ky.get(notificationApiUrl, { headers }).json();
+        } catch (err) {
+          console.error(err);
+          return [];
+        }
+      },
+      default: [],
+      lazy: true
+    }
+  },
+  computed: {
+    unreadCount() {
+      return this.notifications.filter(
+        ({ attributes }) => !JSON.parse(attributes?.READ?.[0] || "true")
+      ).length;
+    },
+    variant() {
+      return this.unreadCount < 1 ? "default" : "danger";
+    }
+  },
+  components: {
+    Dropdown,
+    DropdownHeader,
+    DropdownItem,
+    FontAwesomeIcon,
+    NotificationItem
+  }
+};
+</script>
+
+<!-- Add "scoped" attribute to limit CSS to this component only -->
+<style lang="scss" scoped>
+.notification-icon /deep/ {
```

Comment:
I don't understand what you want to select, is it a willing to select `.notification-icon` both parent html and shadow-dom ? or ? (I don't use deep so...)

Reply:
This selects the root element inside the shadow dom.
scoping the bootstrap styles to the root element and adding a globally unique identifier (guid) to the selector.

`.notification-icon` selects https://github.com/Jasig/NotificationPortlet/blob/3ebfb9a2a3305907e164df8399304099535f916d/notification-portlet-webcomponents/notification-icon/src/components/NotificationIcon.vue#L2
`/deep/` ensures that the guid is applied only to the top selector, and not to child or compound selectors (guid added outside the root breaks several bootstrap selectors). https://vue-loader.vuejs.org/guide/scoped-css.html#deep-selectors

Reply:
Thanks for explanations